PopulateTools/gobierto
# 209 & 1349 - Permit CMS pages & news to have participation layout

Closes [#209](https://github.com/PopulateTools/issues/issues/209)
Closes #1349 

### What does this PR do?

* Splits controllers for `app/controllers/gobierto_participation/pages_controller.rb` and `app/controllers/gobierto_participation/news_controller.rb`, so now pages belonging to participation module can be routed to this controller and inherit the participation layout.
* Fixes the scope of a uniqueness validation in [app/models/gobierto_common/collection.rb](app/models/gobierto_common/collection.rb), `container_id` should be scope in site.
* Fixes the name of the `app/views/gobierto_cms/templates/_new.html.erb` partial. Should be `_news`. This was causing some exceptions.

### How should this be manually tested?

I've done the following testing, but feel free to test other flows:

* Created a collection of news for the Participation module ([http://newalcobendas.gobify.net/admin/collections/167](http://newalcobendas.gobify.net/admin/collections/167))
  * Created a news in this collection and check it has the right layout in the front ([http://newalcobendas.gobify.net/participacion/noticias/noticia-participacion](http://newalcobendas.gobify.net/participacion/noticias/noticia-participacion))
* Created a collection of pages for the Participation module ([http://newalcobendas.gobify.net/admin/collections/166](http://newalcobendas.gobify.net/admin/collections/166))
  * Created a page in this collection and check it has the right layout in the front ([http://newalcobendas.gobify.net/participacion/paginas/pagina-estatica-participacion](http://newalcobendas.gobify.net/participacion/paginas/pagina-estatica-participacion))

### Doubts

I have doubts about the collection of pages Alvaro mentions in the issue ([http://newalcobendas.gobify.net/paginas/acerca-de](http://newalcobendas.gobify.net/paginas/acerca-de)), since now it still has the normal layout. I understand the current behavior is the right one, since the collection is scoped on the site, not participation.

### Future work

I think there's a good opportunity for refactor in the methods in `app/models/gobierto_cms/page.rb` that retrieve pages from a colleciton, module, site etc. Since refactoring this spans to many other files, I've limited myself to adding a few clarifying comments before the methods.

### Does this PR changes any configuration file?

No
